Content.Server/Botany/Systems/MutationSystem.cs

```diff
@@ -138,94 +140,97 @@ public SeedData Cross(SeedData a, SeedData b)
+        // The higher the current value is, the lower the probability of increasing value is, and the higher the probability of decreasive it it.
+        // In other words, it tends to go to the middle.
+        float probIncrease = 1 - (float) valInt / bits;
+        int valIntMutated;
+        if (Random(probIncrease))
         {
-            np = n + 1;
+            valIntMutated = valInt + 1;
         }
         else
         {
-            np = n - 1;
+            valIntMutated = valInt - 1;
         }
 
         // Set value based on mutated thermometer code.
-        float nval = MathF.Min(MathF.Max((float)np/bits * (max - min) + min, min), max);
-        val = nval;
+        float valMutated = Math.Clamp((float) valIntMutated / bits * (max - min) + min, min, max);
+        val = valMutated;
     }
 
-    private void MutateInt(ref int n, int min, int max, int bits, int totalbits, float mult)
+    private void MutateInt(ref int val, int min, int max, int bits, int totalbits, float mult)
     {
-        // Probability that a bit flip happens for this value.
-        float p = mult*bits/totalbits;
-        p = Math.Clamp(p, 0, 1);
-        if (!Random(p))
+        // Probability that a bit flip happens for this value's representation in thermometer code.
+        float probBitflip = mult * bits / totalbits;
+        probBitflip = Math.Clamp(probBitflip, 0, 1);
+        if (!Random(probBitflip))
         {
             return;
         }
 
         // Probability that the bit flip increases n.
-        float p_increase = 1-(float)n/bits;
-        int np;
-        if (Random(p_increase))
+        // The higher the current value is, the lower the probability of increasing value is, and the higher the probability of decreasive it it.
+        // In other words, it tends to go to the middle.
+        float probIncrease = 1 - (float) val / bits;
+        int valMutated;
+        if (Random(probIncrease))
         {
-            np = n + 1;
+            valMutated = val + 1;
         }
         else
         {
-            np = n - 1;
+            valMutated = val - 1;
         }
 
-        np = Math.Min(Math.Max(np, min), max);
-        n = np;
+        valMutated = Math.Clamp(valMutated, min, max);
+        val = valMutated;
     }
 
     private void MutateBool(ref bool val, bool polarity, int bits, int totalbits, float mult)
     {
         // Probability that a bit flip happens for this value.
-        float p = mult * bits / totalbits;
-        p = Math.Clamp(p, 0, 1);
-        if (!Random(p))
+        float prob = mult * bits / totalbits;
+        prob = Math.Clamp(prob, 0, 1);
+        if (Random(prob))
         {
-            return;
+            val = polarity;
         }
-
-        val = polarity;
     }
 
     private void MutateHarvestType(ref HarvestType val, int bits, int totalbits, float mult)
     {
-        float p = mult * bits/totalbits;
-        p = Math.Clamp(p, 0, 1);
-        if (!Random(p))
```

Comment:
The rest of this codebase prefers inverting the if and doing an early return, instead of an `if ()` block as the last statement in a method. *I* of course prefer how it is done here.

Reply:
OK, I will revert it to what it was before.

Reply:
Done.